### PR TITLE
SideCI

### PR DESCRIPTION
**Pull request by ignatenkobrain**

### CODE REVIEW

**Comment by nirbheek** [comment]
What is the advantage of this?

**Comment by ignatenkobrain** [comment]
@nirbheek it will comment inline about newly introduced code errors. For example, https://github.com/rpm-software-management/dnf/pull/683#discussion_r91844830

**Comment by ignatenkobrain** [comment]
@nirbheek why should we worry about this? See https://github.com/mesonbuild/meson/pull/1224 and #1221

**Comment by nirbheek** [comment]
But the check is passing already, so either the CI setup is broken or this won't help with those.

**Comment by nirbheek** [comment]
Ah, it will only warn about new errors?

**Comment by ignatenkobrain** [comment]
@nirbheek for now I disabled it (for PRs), because it will comment each about line-too-long

**Comment by ignatenkobrain** [comment]
@nirbheek yes

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+# W391 blank line at end of file
+# E302 expected 2 blank lines, found 1
+ignore = W391,E302
```
Comment (QuLogic) [inline setup.cfg]: I don't understand why the code base doesn't apply E302. It's so much clearer with two blank lines between classes/functions and the ability to add a blank line within a function.

Comment (nirbheek) [inline setup.cfg]: I wouldn't mind standardising on E302 for our codebase.

Comment (jpakkane) [inline setup.cfg]: I have participated in several attempts in bringing these sorts of "style unifications" into code bases. They have all ended in massive bikeshedding disasters and lost time and energy for everyone involved. The only way to make this work is to use Clang-format (which does not do Python) and have computers do it a) automatically b) reliably. No existing tool can do that for Python code.

Comment (TingPing) [inline setup.cfg]: If a PR simply isn't merged until it passes an automated check it contributors will figure out the rest.

Comment (nirbheek) [inline setup.cfg]: Yeah, our workflow isn't the typical everyone-commits-directly-to-repo workflow, so we could make it work.

Comment (ignatenkobrain) [inline setup.cfg]: @QuLogic @nirbheek removed.

@jpakkane it's not only about code-style, but also about checks for unitialized variables and such (something what me and @TingPing were fixing recently), does clang-format fix the code? ;)

After all, we can tune warnings.

**Comment by jpakkane** [comment]
> it's not only about code-style, but also about checks for unitialized variables and such

These are all 👍. Checking how many pieces of whitespace to put where is not. (the original discussion was about having two linefeeds between classes)

> does clang-format fix the code?

For indentation, yes. For uninitialised variables and such you need clang-tidy.

File: setup.cfg (hunk 1 above)
Comment (TingPing) [inline setup.cfg]: > does clang-format fix the code? 

No clang catches that at the compile step like a good language :P

**Comment by ignatenkobrain** [comment]
@jpakkane let's start from something (all warnings enabled) and then once we encounter some warning we don't care about -- we can disable. After all it's just comments, it's not blocking anything.

**Pull request merged**